src/Windows/Avalonia.Win32/DxgiSwapchain/DxgiConnection.cs

```diff
@@ -0,0 +1,239 @@
+            {
+                Logger.TryGet(LogEventLevel.Error, nameof(DxgiSwapchain))
+                                    ?.Log(this, $"Failed to wait for vblank, Exception: {ex.Message}, HRESULT = {ex.HResult}");
+            }
+
+            while (true)
+            {
+                try
+                {
+                    lock (_syncLock)
+                    {
+                        if (_output is not null)
+                        {
+                            try
+                            {
+                                _output.WaitForVBlank();
+                            }
+                            catch (Exception ex)
+                            {
+                                Logger.TryGet(LogEventLevel.Error, nameof(DxgiSwapchain))
+                                    ?.Log(this, $"Failed to wait for vblank, Exception: {ex.Message}, HRESULT = {ex.HResult}");
+                                _output.Dispose();
+                                _output = null;
+                                GetBestOutputToVWaitOn();
+                            }
+                        }
+                        else
+                        {
+                            // well since that obviously didn't work, then let's use the lowest-common-denominator instead 
+                            // for reference, this has never happened on my machine,
+                            // but theoretically someone could have a weirder setup out there 
+                            DwmFlush();
+                        }
+                        Tick?.Invoke(_stopwatch.Elapsed);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Logger.TryGet(LogEventLevel.Error, nameof(DxgiSwapchain))
+                                    ?.Log(this, $"Failed to wait for vblank, Exception: {ex.Message}, HRESULT = {ex.HResult}");
+                }
+            }
+        }
+
+        // Note: Defining best as display with highest refresh rate on 
+        private void GetBestOutputToVWaitOn()
+        {
+            double highestRefreshRate = 0.0d;
+            HRESULT retval = default;
+            IDXGIFactory? fact = null;
+            void* factPointer = null;
+
+            // IDXGIFactory Guid: [Guid("7B7166EC-21C7-44AE-B21A-C9AE321AE369")]
+            Guid factoryGuid = MicroComRuntime.GetGuidFor(typeof(IDXGIFactory));
+            retval = CreateDXGIFactory(&factoryGuid, (void**)&factPointer);
+            if (retval.FAILED)
+            {
+                // To be clear, if this fails then it means we've tried to use dxgi methods on a system that does not support any of them 
+                // I fully expect that if we hit this, then the application is done for. I don't think ANGLE would work either, frankly. 
+                throw new Win32Exception((int)retval);
+            }
+
+            fact = MicroComRuntime.CreateProxyFor<IDXGIFactory>(factPointer, true);
```

Comment:
Temporary MicroCom-wrapped objects should be wrapped with a `using` keyword to ensure correct disposal on time. If you need to conditionally retain a reference in a field, use `CloneReference()`, it will call `AddRef()` under the hood.

Reply:
Several using statements have been added to DxgiConnection and DxgiRenderTarget, the CloneReference method has been used as required.